src/features/trigger/CronTrigger.tsx

```diff
@@ -1,34 +1,49 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { useDispatch } from 'react-redux'
 
 import Icon from '../../chrome/Icon'
-import Link from '../../chrome/Link'
 import { WorkflowTrigger } from '../../qrimatic/workflow'
-import Block from '../workflow/Block'
-import { showModal } from '../app/state/appActions'
-import { ModalType } from '../app/state/appState'
-import { parseHourlyStart, parseDailyStart,  } from './util'
+import { parseHourlyStart, parseDailyStart } from './util'
+import CronTriggerEditor from "./CronTriggerEditor";
+import { changeWorkflowTrigger, deleteWorkflowTrigger } from "../workflow/state/workflowActions";
 
 export interface CronTriggerProps {
   trigger: WorkflowTrigger
+  onCreateDelete: () => void
+  editMode?: boolean
 }
 
 const CronTrigger: React.FC<CronTriggerProps> = ({
-  trigger
+  trigger: initialTrigger,
+  onCreateDelete,
+  editMode
 }) => {
-
   const dispatch = useDispatch()
 
+  // manage a collection of triggers in local state until they are "saved"
+  const [ triggers, setTriggers ] = useState<WorkflowTrigger[]>([initialTrigger])
+  const [ isEditable, setIsEditable ] = useState(!!editMode)
   // on edit, open AddTriggerModal and pass in the correct type and array of WorkflowTrigger
-  const handleEditClick = () => {
-    dispatch(showModal(ModalType.addTrigger, {
-      type: 'cron',
-      triggers: [ trigger ]
-    }))
+
+  const handleSaveTrigger = () => {
+    setIsEditable(false)
+    onCreateDelete()
+    dispatch(changeWorkflowTrigger(0, triggers[0]))
+  }
+
+  const handleTriggerDelete = () => {
+    setIsEditable(false)
+    onCreateDelete()
+    dispatch(deleteWorkflowTrigger(0))
+
+  }
+
+  const handleTriggerChange = (t: WorkflowTrigger) => {
+    setTriggers([t])
   }
 
   // split the ISO8601 repeating interval (R/{starttime}/{interval})
-  const [, startTime, interval] = trigger.periodicity.split('/')
+  const [, startTime, interval] = triggers[0].periodicity.split('/')
```

Comment:
I'm getting typescript error on the `periodicity` field.

`triggers` is an array of `WorkflowTrigger`. The `WorkflowTrigger` interface does not have `periodicity` as a properly. We need to first cast `trigger[0]` as a `CronTrigger` before trying to get `periodicity` from it:

```
 const cronTrigger = triggers[0] as CronTrigger
 const [, startTime, interval] = cronTrigger.periodicity.split('/')
```